diff --git a/lib_pypy/_curses.py b/lib_pypy/_curses.py
--- a/lib_pypy/_curses.py
+++ b/lib_pypy/_curses.py
@@ -6,6 +6,13 @@
     raise ImportError('No module named _curses')
 from functools import wraps
 
+# SNAP: Use bundled terminfo data. The host's may be incompatible
+import os
+if 'TERMINFO_DIRS' not in os.environ:
+    snapdir = os.path.dirname(os.path.dirname(__file__))
+    os.environ['TERMINFO_DIRS'] = os.path.join(snapdir, 'lib/terminfo')
+    del snapdir
+
 from _curses_cffi import ffi, lib
 
 version = b"2.2"
diff --git a/lib_pypy/pyrepl/curses.py b/lib_pypy/pyrepl/curses.py
--- a/lib_pypy/pyrepl/curses.py
+++ b/lib_pypy/pyrepl/curses.py
@@ -25,6 +25,14 @@
 import sys
 if '__pypy__' in sys.builtin_module_names:
     # pypy case
+
+    # SNAP: Use bundled terminfo data. The host's may be incompatible
+    import os
+    if 'TERMINFO_DIRS' not in os.environ:
+        snapdir = os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
+        os.environ['TERMINFO_DIRS'] = os.path.join(snapdir, 'lib/terminfo')
+        del snapdir
+
     import _minimal_curses as _curses
 else:
     # cpython case
